src/api-service/__app__/onefuzzlib/notifications/main.py

```diff
@@ -97,6 +97,34 @@ def get_notifications(container: Container) -> List[Notification]:
     return Notification.search(query={"container": [container]})
 
 
+def get_regression_report_task(report: RegressionReport) -> Optional[Task]:
+    # crash_test_result is required, but report & no_repro are not
+    if report.crash_test_result.crash_report:
+        return Task.get(
+            report.crash_test_result.crash_report.job_id,
+            report.crash_test_result.crash_report.task_id,
+        )
+    if report.crash_test_result.no_repro:
+        return Task.get(
+            report.crash_test_result.no_repro.job_id,
+            report.crash_test_result.no_repro.task_id,
+        )
+
+    # original_crash_test_result is not required, nor are report & no_repro.
+    if report.original_crash_test_result:
+        if report.original_crash_test_result.crash_report:
+            return Task.get(
+                report.original_crash_test_result.crash_report.job_id,
```

Comment:
One thing I don't follow here: I would expect this to be the `job_id`/`task_id` for the task that _originally_ found crash that has now regressed. That means here we'd be returning a totally different (past) task, compared to the other cases. Am I mistaken?

Reply:
By using the current `job_id` and `task_id`, we provide the metadata related to the task that added the regression.  This includes the setup containers and tags related to the regression.

Reply:
My expectation: we want to use the _current_ job ID/task ID.

My reading of the code in this branch: in this branch, but not the other branches we are extracting and returning the _old_ job ID/task ID, which is not what we want.

Reply:
I believe I understand your concern.

In the case the new regression report is missing both a crash report and a norepo, which shouldn't happen but isn't protected against, this will fall back to identifying the previous task.

I think it's a toss up, but I leaned toward providing the old data rather than no data, given the context.